src/engine/channels/enginedeck.cpp

```diff
@@ -206,6 +207,18 @@ void EngineDeck::cloneStemState(const EngineDeck* deckToClone) {
     VERIFY_OR_DEBUG_ASSERT(deckToClone) {
         return;
     }
+    // Sampler and preview decks don't have stem controls
+    if ((m_stemGain.empty() && m_stemMute.empty()) ||
+            (deckToClone->m_stemGain.empty() &&
+                    deckToClone->m_stemMute.empty())) {
+        return;
+    }
```

Comment:
The verify blow includes this check and sticks around at release time. So this check can be removed.

Reply:
It does but it's not equivalent. 

Having the two vector empty is an expected case, for which we do an early return. The assert condition adds extra usecase that would crash the for-loop but are not expected usecase on which we should assert.

I agree that this would technically add couple of extra instructions at release time, but since this function is not time critical, I would suggest we keep the clear distinction between the early return on excepted usecase, and protective return on unexpected case.

Reply:
Ah I see. How about replacing the first loop by 
`If (isPrimaryDeck())`?

Reply:
That sounds much cleaner indeed. Updated.